numpy/_core/src/_simd/_simd.c

```diff
@@ -88,13 +79,31 @@ PyMODINIT_FUNC PyInit__simd(void)
     NPY_MTARGETS_CONF_DISPATCH(NPY_CPU_HAVE, ATTACH_MODULE, MAKE_MSVC_HAPPY)
     NPY_MTARGETS_CONF_BASELINE(ATTACH_BASELINE_MODULE, MAKE_MSVC_HAPPY)
 
-#if Py_GIL_DISABLED
-    // signal this module supports running with the GIL disabled
-    PyUnstable_Module_SetGIL(m, Py_MOD_GIL_NOT_USED);
+    return 0;
+err:
+    return -1;
+}
+
+static struct PyModuleDef_Slot _simd_slots[] = {
+    {Py_mod_exec, _simd_exec},
+#if PY_VERSION_HEX >= 0x030c00f0  // Python 3.12+
+    {Py_mod_multiple_interpreters, Py_MOD_MULTIPLE_INTERPRETERS_NOT_SUPPORTED},
+#endif
+#if PY_VERSION_HEX >= 0x030d00f0  // Python 3.13+
+    // signal that this module supports running without an active GIL
+    {Py_mod_gil, Py_MOD_GIL_NOT_USED},
 #endif
+    {0, NULL},
+};
 
-    return m;
-err:
-    Py_DECREF(m);
-    return NULL;
+static struct PyModuleDef defs = {
+    .m_base = PyModuleDef_HEAD_INIT,
+    .m_name = "numpy._core._simd",
+    .m_size = 0,
```

Comment:
This was -1, did it become 0 on purpose?

Reply:
Yes, I think we should change it to 0 in a future PR. Please only change it in a single  PR, lets get one of these in before the mass move of all the other ones. Does the change impact performance? I would suggest redoing this in #29030, or one of the others that touch a commonly used module, and then running the benchmarks to see if there is a regression when using multi-phase init.

Reply:
Ok, I've updated #29030 to keep `m_size = -1`.

Is there infrastructure to benchmark a PR? My laptop isn't an ideal environment & I don't have a spare computer to run the [local benchmarks](https://numpy.org/doc/stable/benchmarking.html).

Reply:
From https://github.com/numpy/numpy/actions/runs/15179012580/job/42684537494?pr=29030

> m_size may not be negative for multi-phase initialization

I'll get the Python documentation updated, this isn't very obvious!

In the meantime, I'll revert the change to #29030.